### PR TITLE
SIGTERM node shutdown type

### PR DESCRIPTION
**Pull request by stu-elastic**
Adds the SIGTERM shutdown type which will be used for graceful shutdown in response to SIGTERM.   Right now it's the same as REPLACE.

### CODE REVIEW

**Review by AthenaEryma — CHANGES_REQUESTED** [review]
Mostly looks good, left a couple nitpicks and one concern about serialization.

File: server/src/main/java/org/elasticsearch/cluster/metadata/SingleNodeShutdownMetadata.java [hunk 1]
```diff
@@ -380,7 +380,8 @@ public SingleNodeShutdownMetadata build() {
     public enum Type {
         REMOVE,
         RESTART,
-        REPLACE;
+        REPLACE,
+        SIGTERM; // local version of REMOVE
```
[the remark this answers is not in the document]
Comment (stu-elastic) [reply server/src/main/java/org/elasticsearch/cluster/metadata/SingleNodeShutdownMetadata.java]: Updated.

**Review by AthenaEryma — APPROVED** [review]
LGTM, thanks!

**Comment by stu-elastic** [comment]
Updated to `TransportVersion.V_8_9_0`

**Pull request merged**